dev-util/jenkins-bin/files/jenkins-bin.confd

```diff
@@ -60,3 +60,15 @@ JENKINS_PIDFILE="/var/run/jenkins.pid"
 # War File
 #
 JENKINS_WAR="/opt/jenkins/jenkins.war"
+
+#
+# Java customization
+#
+JAVA_HOME=/usr/lib64/openjdk-11
```

Comment:
What is the reason for hardcoding a specific java version here? Currently jenkins uses the system-vm version and that seems to work fine? I'd also rather not hardcode a specific version if multiple versions work (e.g. 11 and 17).

Reply:
> What is the reason for hardcoding a specific java version here? Currently jenkins uses the system-vm version and that seems to work fine? I'd also rather not hardcode a specific version if multiple versions work (e.g. 11 and 17).

Hello @graaff 
That time it doesn't work with the icedtea-8 release that was the default one for system-vm.
I need to install openjdk-11 that was required for multiple Jenkins plugins and the only way to do that was with this approach.
I also found this kind of approach in other Java related packages. The only missing point is that I could improve it as part of a use flag related to the Java requirement, instead of being hardcoded as you pointed out.

Reply:
The ebuild now requires either jre:17 or jre:11 (and both versions work with Jenkins). This assumes that the system java is one of those, which is not necessarily the case.

Reply:
I am currently in a hurry, so I can only comment briefly. But I hope you get the idea.

Hardcoding a specific java version is usually not required.

```
$ depend-java-query --get-vm=virtual/jre-19
openjdk-bin-19
```

queries the system for a suitable JVM. Details about the returned JVM can be resolved via java-config. For example

```bash
$ java-config -P openjdk-bin-19
# Autogenerated by java-config
# Java Virtual Machine: Eclipse Temurin JDK 19.0.2_p7
JAVA_HOME=/opt/openjdk-bin-19.0.2_p7
JDK_HOME=/opt/openjdk-bin-19.0.2_p7
JAVAC=/opt/openjdk-bin-19.0.2_p7/bin/javac
PATH=/opt/openjdk-bin-19.0.2_p7/bin
ROOTPATH=/opt/openjdk-bin-19.0.2_p7/bin
LDPATH=/opt/openjdk-bin-19.0.2_p7/lib/:/opt/openjdk-bin-19.0.2_p7/lib/server/
MANPATH=
```

Reply:
@Flowdalic So, for example, if someone wants to use openjdk-17 with Jenkins, but his default system-vm is IcedTea JDK, that means that the initialization script for Jenkins service should use java-config command to query for a supported Java VM and load it to the environment?
Don't know if there is any eclass that should help with this kind of task, something that I'm looking forward for...

Reply:
I've finally had the time to look more into our java tooling. Ideally jenkins-bin ebuild would use `java-pkg_dolauncher` from `java-utils-2.eclass`. This would install a launcher which would perform basically the steps I've mentioned above. I think this may requires also calling `jav_pkg_dowar` or maybe `java_pkg_dojar`, or maybe in your case `java-pkg_addcp /opt/jenkins`.

Reply:
Thank you @Flowdalic for the advise. I'm going to do that.